mixer/adapter/stdio/stdio.go

```diff
@@ -117,6 +118,15 @@ func (h *handler) mapSeverityLevel(severity string) zapcore.Level {
 	return level
 }
 
+func convertSomeTypestoStringValue(value interface{}) interface{} {
+	switch vt := value.(type) {
+	case []byte:
+		return interface{}(net.IP(vt).String())
```

Comment:
This is not correct. []byte can be anything, not just an IP address.

To have a log value be printed as an IP address, the byte[] needs to be converted to an IP address in the user's config instead and passed to the adapter as a value of type IP_ADDRESS.

Reply:
I think we need a `string` function that is compiled to ip_string. The user type IP is erased in the attribute bag, so we cannot distinguish between []byte and IP address from dynamic values.

Reply:
The config is using an IP_ADDRESS attribute, correct? Why do we need something else in the config?

This seems like a mixer-internal issue. Should values taken from those kinds of attributes should be treated as `net.IP` instead of `[]byte`?  Even our existing `ip()` func converts first to `net.IP` then casts to `[]byte`.

Reply:
In adapter.Stringify function https://sourcegraph.com/github.com/istio/istio@master/-/blob/mixer/pkg/adapter/instanceUtil.go#L34:6 also, we are assuming []byte is going to be an IP Address..

Reply:
We can extend the set of runtime values to have `net.IP`. But we need to update every consumer of the runtime values (including the generated code that does type switch checks). That just seems a more involved solution.

TBH I don't think there is any use case for []byte that is not IP right now.

Reply:
can we cheat here, and test to see if the `[]byte` parses as an IP ?

```
       if ip := net.ParseIP(string(vt[:])); ip != nil {
		return ip.String()
       }
       return vt
```

Reply:
Thanks @douglas-reid for the idea. string(vt[:]) doesn't really convert it into string format that ParseIP accepts but am using To16 function of IP that validates its a valid ip address...